postgresqlPackages: fix update scripts with PG versioning scheme

wolfgangwalther opened the pull request:
PostgreSQL itself and some of its extensions use a `REL_X_Y_Z` versioning scheme, or similar - delimiting by `_` instead of `.`.

This fixes the update scripts for those packages. Depends on:
- [x] https://github.com/Mic92/nix-update/pull/401

## Things done

- [x] Tested by actually updating those 3 packages.
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md), [pkgs/README.md](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md), [maintainers/README.md](https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md) and other contributing documentation in corresponding paths.

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

wolfgangwalther committed:
postgresqlPackages.pg_squeeze: fix update script and version

wolfgangwalther committed:
postgresqlPackages.plr: fix update script and version

wolfgangwalther committed:
postgresqlPackages.wal2json: fix update script and version

wolfgangwalther committed:
psqlodbc: fix update script

wolfgangwalther committed:
psqlodbc: refactor

philiptaron commented on pkgs/by-name/ps/psqlodbc/package.nix:
```diff
@@ -56,11 +49,22 @@ stdenv.mkDerivation rec {
-  meta = with lib; {
+  passthru =
+    {
+      updateScript = nix-update-script {
+        extraArgs = [ "--version-regex=^REL-(\\d+)_(\\d+)_(\\d+)$" ];
+      };
```
This can be whatever we want, right? Like, say, `passthru.updateScript = ./update.sh`?

philiptaron reviewed (commented):
I do like it better.

wolfgangwalther committed:
postgresqlPackages.pg_squeeze: 1.7.0 -> 1.9.0

Release notes:
https://github.com/cybertec-postgresql/pg_squeeze/releases/tag/REL1_8_0
https://github.com/cybertec-postgresql/pg_squeeze/releases/tag/REL1_9_0

wolfgangwalther committed:
postgresqlPackages.plr: 8.4.7 -> 8.4.8

Changelog:
https://github.com/postgres-plr/plr/blob/master/changelog.md#848---2025-05-17
